packages/x-tree-view/src/RichTreeView/RichTreeView.tsx

```diff
@@ -0,0 +1,323 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { styled, useThemeProps } from '@mui/material/styles';
+import composeClasses from '@mui/utils/composeClasses';
+import { useSlotProps } from '@mui/base/utils';
+import { getRichTreeViewUtilityClass } from './richTreeViewClasses';
+import { RichTreeViewProps } from './RichTreeView.types';
+import { useTreeView } from '../internals/useTreeView';
+import { TreeViewProvider } from '../internals/TreeViewProvider';
+import { DEFAULT_TREE_VIEW_PLUGINS } from '../internals/plugins';
+import { TreeItem, TreeItemProps } from '../TreeItem';
+import { buildWarning } from '../internals/utils/warning';
+
+const useUtilityClasses = <R extends {}, Multiple extends boolean | undefined>(
+  ownerState: RichTreeViewProps<R, Multiple>,
+) => {
+  const { classes } = ownerState;
+
+  const slots = {
+    root: ['root'],
```

Comment:
Shouldn't this also have the `item` slot and pass the relevant item class to all the items? 🤔

Reply:
We need to discuss the various customization API we want to put in place for the item.
I do agree that an `item` class at the Tree View level would probably we a good and easy way to style items.

Reply:
Makes sense. 👍